run-platform.sh

```diff
@@ -126,21 +126,40 @@ parse_args() {
   debug "OPTION only_env: $opt_only_env"
   debug "OPTION only_pull: $opt_only_pull"
   debug "OPTION build_local: $opt_build_local"
-  debug "OPTION upgrade: $opt_upgrade"
+  debug "OPTION upgrade: $opt_update"
   debug "OPTION verbose: $opt_verbose"
   debug "OPTION version: $opt_version"
 }
 
 do_git_pull() {
-  if [ "$opt_upgrade" = false ]; then
+  if [ "$opt_update" = false ]; then
     return
   fi
 
-  echo -e "Performing git switch to ""$blue_text""main branch""$default_text".
-  git switch main
+  echo "Fetching release tags."
+  git fetch
 
-  echo -e "Performing ""$blue_text""git pull""$default_text"" on main branch."
-  git pull
+  if [[ "$opt_version" == "latest" ]]; then
+    branch=`git describe --tags --abbrev=0`
```

Comment:
@hari-kuriakose tried this from my feature branch 
![image](https://github.com/user-attachments/assets/83dd3fc1-dca2-4c00-b2fb-a7679dd0e1a7)
and from main (without a `git pull`)
![image](https://github.com/user-attachments/assets/6d06b2db-b394-4c4c-a519-9584689291cf)
Result from main after a `git pull`
![image](https://github.com/user-attachments/assets/42a90038-292e-476b-a2ae-8aeffd9064ce)


The version selected in the first 2 cases was `v0.73.0`
1. It didn't select the latest tag of `v0.74.0` without a `git pull`
1. It didn't select the patch version releases of `v0.73.2`

Is this the intended behaviour of the `latest` option now?

Reply:
@chandrasekharan-zipstack Good find.

`git fetch` by default only pulls those tags reference by the current branch. Fixed this by explicitly fetching all tags always via `git fetch --quiet --tags`.